### PR TITLE
Added debug log for evictions from registry after ttl expires

### PR DESCRIPTION
**Pull request by soberpeach**
### What does this PR do?

Adds a debug level log whenever a file is evicted from the registry.

### Motivation

This would make it easier to debug support cases where log files age out of the registry because their TTL is shorter than their log production frequency. The current default TTL for log files is 23 hours, so the agent will evict any log files from any app that produces logs less frequently than every 23 hours, causing the whole file to be read again when the agent picks it up.

### Additional Notes

N/A

### Possible Drawbacks / Trade-offs

N/A

### Describe how to test/QA your changes

1. Set `logs_config.auditor_ttl` to a very low value (I used 0.001)
2. Set `log_level` to debug
3. Run the agent with `logs_enabled` set to true
4. Check the agent logs to see if it's emitting the proper logs noting log file eviction.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/add-debug-log-for-files-evicted-from-registry-9811327329dbca7e.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Add debug level logs when files are evicted from ```registry.log```
```
Comment (gh123man) [inline releasenotes/notes/add-debug-log-for-files-evicted-from-registry-9811327329dbca7e.yaml]:
```suggestion
    Add debug level logs when files are evicted from ```registry.json```
```

**Review by gh123man — APPROVED** [review]
LGTM!

**Review by rayz — APPROVED** [review]

**Review by cswatt — APPROVED** [review]
release notes LGTM

**Pull request merged**